src/graphql/thegraph/PoolData.ts

```diff
@@ -0,0 +1,64 @@
+import { ApolloClient, InMemoryCache } from '@apollo/client'
+import { ChainId } from '@uniswap/sdk-core'
+import gql from 'graphql-tag'
+import { useMemo } from 'react'
+
+import { usePoolDataQuery } from './__generated__/types-and-hooks'
+import { CHAIN_SUBGRAPH_URL } from './apollo'
+
+gql`
+  query PoolData($poolId: [ID!]) {
+    data: pools(where: { id_in: $poolId }, orderBy: totalValueLockedUSD, orderDirection: desc, subgraphError: allow) {
+      id
+      feeTier
+      liquidity
+      sqrtPrice
+      tick
+      token0 {
+        id
+        symbol
+        name
+        decimals
+        derivedETH
+      }
+      token1 {
+        id
+        symbol
+        name
+        decimals
+        derivedETH
+      }
+      token0Price
+      token1Price
+      volumeUSD
+      volumeToken0
+      volumeToken1
+      txCount
+      totalValueLockedToken0
+      totalValueLockedToken1
+      totalValueLockedUSD
+    }
+    bundles(where: { id: "1" }) {
+      ethPriceUSD
+    }
+  }
+`
+
+export function usePoolData(poolAddress: string, chainId?: ChainId) {
+  const poolId = [poolAddress]
+  const apolloClient = useMemo(
+    () =>
+      new ApolloClient({
+        cache: new InMemoryCache(),
+        uri: CHAIN_SUBGRAPH_URL[chainId ?? ChainId.MAINNET],
+      }),
+    [chainId]
+  )
+  const { data, loading } = usePoolDataQuery({ variables: { poolId }, client: apolloClient })
+  return useMemo(() => {
+    return {
+      data: data?.data[0],
```

Comment:
this feels a little jank

Reply:
because of the double data or taking the first elem? This does handle invalid poolIds by returning `undefined` for data. ex: https://interface-git-cab-initpdp-uniswap.vercel.app/#/pools/ethereum/invalid

Reply:
`data: data?.data[0]` 

both reasons I guess. When I try to verbalize this I get something like: "data is data's data array's first element, except if data is undefined, in which case we return undefined data". I think it would be better to assign the query data to a descriptive var name, then handle the undefined case separately before returning the result in the useMemo.